common/src/from_server_interface.rs

```diff
@@ -0,0 +1,279 @@
+}
+
+impl TryFrom<FromServer> for proto::FromServer {
+    type Error = &'static str;
+
+    fn try_from(item: FromServer) -> Result<Self, Self::Error> {
+        match item {
+            FromServer::UpdateWorkload(ankaios) => Ok(proto::FromServer {
+                from_server_enum: Some(proto::from_server::FromServerEnum::UpdateWorkload(
+                    proto::UpdateWorkload {
+                        added_workloads: ankaios
+                            .added_workloads
+                            .into_iter()
+                            .map(|x| x.into())
+                            .collect(),
+                        deleted_workloads: ankaios
+                            .deleted_workloads
+                            .into_iter()
+                            .map(|x| x.into())
+                            .collect(),
+                    },
+                )),
+            }),
+            FromServer::UpdateWorkloadState(ankaios) => Ok(proto::FromServer {
+                from_server_enum: Some(proto::from_server::FromServerEnum::UpdateWorkloadState(
+                    proto::UpdateWorkloadState {
+                        workload_states: ankaios
+                            .workload_states
+                            .iter()
+                            .map(|x| x.to_owned().into())
+                            .collect(),
+                    },
+                )),
+            }),
+            FromServer::Response(ankaios) => Ok(proto::FromServer {
+                from_server_enum: Some(proto::from_server::FromServerEnum::Response(
+                    proto::Response {
+                        request_id: ankaios.request_id,
+                        response_content: Some(ankaios.response_content.into()),
+                    },
+                )),
+            }),
+            FromServer::Stop(_) => Err("Stop command not implemented in proto"),
+        }
+    }
+}
+
+#[async_trait]
+pub trait AgentInterface {
+    async fn update_workload(
+        &self,
+        added_workloads: Vec<WorkloadSpec>,
+        deleted_workloads: Vec<DeletedWorkload>,
+    ) -> Result<(), FromServerInterfaceError>;
+    async fn update_workload_state(
+        &self,
+        workload_running: Vec<WorkloadState>,
+    ) -> Result<(), FromServerInterfaceError>;
+    async fn complete_state(
+        &self,
+        request_id: String,
+        complete_state: commands::CompleteState,
+    ) -> Result<(), FromServerInterfaceError>;
+    async fn success(&self, request_id: String) -> Result<(), FromServerInterfaceError>;
+    async fn error(
+        &self,
+        request_id: String,
+        error: commands::Error,
+    ) -> Result<(), FromServerInterfaceError>;
+    async fn stop(&self) -> Result<(), FromServerInterfaceError>;
+}
+
+pub type FromServerSender = tokio::sync::mpsc::Sender<FromServer>;
+pub type FromServerReceiver = tokio::sync::mpsc::Receiver<FromServer>;
+
+#[async_trait]
+impl AgentInterface for FromServerSender {
```

Comment:
As we have only renamed this trait in this PR, I would exclude this from this PR, as I want this PR to be merged as soon as possible, to prevent more merge conflicts. We can handle this in a separate PR.

Reply:
I have opened an issue. https://github.com/eclipse-ankaios/ankaios/issues/168